File: backend/app/services/appointment_service.py
from uuid import UUID

from fastapi import HTTPException
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.models.appointment import Appointment
from app.schemas.appointment import AppointmentCreate


class AppointmentService:

    @staticmethod
    def create_appointment(
        db: Session,
        appointment: AppointmentCreate,
    ):

        db_appointment = Appointment(
            customer_id=appointment.customer_id,
            employee_id=appointment.employee_id,
            service_id=appointment.service_id,
            appointment_date=appointment.appointment_date,
        )

        db.add(db_appointment)
        db.commit()
        db.refresh(db_appointment)

        return db_appointment

    @staticmethod
    def list_appointments(
        db: Session,
    ):

        return db.scalars(
            select(Appointment)
        ).all()

    @staticmethod
    def get_appointment(
        db: Session,
        appointment_id: UUID,
    ):

        appointment = db.get(
            Appointment,
            appointment_id,
        )

        if appointment is None:
            raise HTTPException(
                status_code=404,
                detail="Appointment not found",
            )

        return appointment

    @staticmethod
    def delete_appointment(
        db: Session,
        appointment_id: UUID,
    ):

        appointment = AppointmentService.get_appointment(
            db,
            appointment_id,
        )

        db.delete(appointment)

        db.commit()

File: frontend/src/services/appointmentService.js
import api from "./api";

export async function getAppointments() {

    const response = await api.get("/appointments");

    return response.data;

}

export async function createAppointment(appointment) {

    const response = await api.post(
        "/appointments",
        appointment
    );

    return response.data;

}

export async function updateAppointment(appointmentId, appointment) {

    const response = await api.put(
        `/appointments/${appointmentId}`,
        appointment
    );

    return response.data;

}

export async function deleteAppointment(appointmentId) {

    await api.delete(
        `/appointments/${appointmentId}`
    );

}